relayer/relayer/src/connections/azero.rs

```diff
@@ -0,0 +1,179 @@
+    SignerError(#[from] signer_client::Error),
+}
+
+impl AzeroSignerClient {
+    async fn new(cid: u32, port: u32) -> Result<Self, Error> {
+        let res = spawn_blocking(move || {
+            let client = Client::new(cid, port)?;
+            let account_id = client.account_id()?;
+            let client = Arc::new(Mutex::new(client));
+
+            Ok::<_, signer_client::Error>(Self { client, account_id })
+        })
+        .await??;
+
+        Ok(res)
+    }
+}
+
+enum AzeroSigner {
+    Dev(Box<KeyPair>),
+    Signer(AzeroSignerClient),
+}
+
+impl AzeroSigner {
+    fn account_id(&self) -> &AccountId {
+        match self {
+            AzeroSigner::Dev(keypair) => keypair.account_id(),
+            AzeroSigner::Signer(signer) => &signer.account_id,
+        }
+    }
+
+    async fn sign(&self, payload: &[u8]) -> Result<MultiSignature, anyhow::Error> {
+        match self {
+            AzeroSigner::Dev(keypair) => Ok(keypair.signer().sign(payload).into()),
+            AzeroSigner::Signer(signer) => {
+                let client = signer.client.clone();
+                let payload = payload.to_vec();
+
+                let res = spawn_blocking(move || {
+                    let client = client.lock().unwrap();
+                    let signature = client.sign(&payload)?;
+                    Ok::<_, signer_client::Error>(signature)
+                })
+                .await??;
+
+                Ok(res)
+            }
+        }
+    }
+}
+
+pub struct AzeroConnectionWithSigner {
+    connection: AzeroWsConnection,
+    signer: AzeroSigner,
+}
+
+impl AzeroConnectionWithSigner {
+    pub async fn with_signer(
+        connection: AzeroWsConnection,
+        cid: u32,
+        port: u32,
+    ) -> Result<Self, Error> {
+        let client = AzeroSignerClient::new(cid, port).await?;
+        let signer = AzeroSigner::Signer(client);
+        Ok(Self { connection, signer })
+    }
+
+    pub fn with_keypair(connection: AzeroWsConnection, keypair: KeyPair) -> Self {
+        let signer = AzeroSigner::Dev(Box::new(keypair));
+        Self { connection, signer }
+    }
+}
+
+impl AsConnection for AzeroConnectionWithSigner {
+    fn as_connection(&self) -> &Connection {
+        &self.connection
+    }
+}
+
+#[async_trait::async_trait]
+impl SignedConnectionApi for AzeroConnectionWithSigner {
+    async fn send_tx<Call: TxPayload + Send + Sync>(
+        &self,
+        tx: Call,
+        status: TxStatus,
+    ) -> anyhow::Result<TxInfo> {
+        self.send_tx_with_params(tx, Default::default(), status)
+            .await
+    }
+
+    async fn send_tx_with_params<Call: TxPayload + Send + Sync>(
+        &self,
+        tx: Call,
+        params: ParamsBuilder,
+        status: TxStatus,
+    ) -> anyhow::Result<TxInfo> {
+        if let Some(details) = tx.validation_details() {
+            info!(target:"aleph-client", "Sending extrinsic {}.{} with params: {:?}", details.pallet_name, details.call_name, params);
```

Comment:
target is confusing in this log, since it's clearly not coming from the aleph-client but from relayer?

Reply:
:+1: